bukuserver/api.py

```diff
@@ -83,23 +85,59 @@ def search_tag(
     return list(sorted(tags)), dict(counter.most_common(limit))
 
 
-class ApiTagView(MethodView):
+def _fetch_data(convert):
+    form = ApiFetchDataForm(request.form)
+    if not form.validate():
+        return Response.INPUT_NOT_VALID(data={'errors': form.errors})
+    try:
+        return Response.SUCCESS(data=convert(buku.fetch_data(form.url.data)))
+    except Exception as e:
+        current_app.logger.debug(str(e))
+        return Response.FAILURE()
+
+def handle_network():
+    return _fetch_data(lambda x: {'title': x.title, 'description': x.desc, 'tags': x.keywords,
+                                  'recognized mime': int(x.mime), 'bad url': int(x.bad)})
+
+def fetch_data():
+    return _fetch_data(lambda x: x._asdict())
+
+
+def refresh_bookmark(index: T.Optional[int]):
+    bdb = getattr(flask.g, 'bukudb', get_bukudb())
+    if index and not bdb.get_rec_by_id(index):
+        return Response.BOOKMARK_NOT_FOUND()
+    result_flag = bdb.refreshdb(index or None, request.form.get('threads', 4))
+    return Response.from_flag(result_flag)
+
+
+get_tiny_url = lambda index: Response.REMOVED()
 
-    def get(self, tag: T.Optional[str]):
+
+def get_all_tags():
+    return Response.SUCCESS(data={"tags": search_tag(db=get_bukudb(), limit=5)[0]})
```

Comment:
Separated collection handlers from collection item handlers – mostly because it's cleaner this way